Toolkit/ArcGISToolkit/JobManager.swift

```diff
@@ -222,26 +217,27 @@ public class JobManager: NSObject {
      `func application(_ application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping`
      */
     public func application(application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
-        if self.jobs.count > 0{
-            self.checkStatusForAllJobs{ completedWithoutErrors in
-                if completedWithoutErrors{
-                    completionHandler(.newData)
-                }
-                else{
-                    completionHandler(.failed)
-                }
-            }
+        guard jobs.count > 0 else {
+            return completionHandler(.noData)
         }
-        else{
-            completionHandler(.noData)
+        
+        checkStatusForAllJobs { completedWithoutErrors in
+            if completedWithoutErrors {
+                completionHandler(.newData)
+            }
+            else{
+                completionHandler(.failed)
+            }
         }
     }
 
     
     /// Resume all paused and not-started jobs.
-    public func resumeAllPausedJobs(statusHandler: @escaping JobStatusHandler, completion: @escaping JobCompletionHandler){
-        keyedJobs.filter{ $0.1.status == .paused || $0.1.status == .notStarted}.forEach{
-            $0.1.start(statusHandler: statusHandler, completion:completion)
+    public func resumeAllPausedJobs(statusHandler: @escaping JobStatusHandler, completion: @escaping JobCompletionHandler) {
+        // Note that a job is also paused if it is created from JSON. So any jobs that have been stored in
+        // UserDefaults and reloaded will be in the paused state.
+        keyedJobs.filter({ $0.value.status == .paused || $0.value.status == .notStarted }).forEach {
```

Comment:
Lazily filtering would prevent looping twice:

```suggestion
        keyedJobs.lazy.filter({ $0.value.status == .paused || $0.value.status == .notStarted }).forEach {
```

Reply:
Ooh. I was not aware of LazyCollections. Thank you!